cloudinit/config/cc_set_passwords.py

```diff
@@ -184,11 +143,22 @@ def handle_ssh_pwauth(pw_auth, distro: Distro):
         LOG.debug("No need to restart SSH service, %s not updated.", cfg_name)
         return
 
-    if restart_ssh:
-        distro.manage_service("restart", service)
-        LOG.debug("Restarted the SSH daemon.")
+    if distro.uses_systemd():
+        state = subp.subp(
+            f"systemctl show --property ActiveState --value {service}"
+        ).stdout
+        if state in ["active", "activating"]:
```

Comment:
Think we should restart if in the `reloading` state?

Reply:
Properties [list](https://www.freedesktop.org/software/systemd/man/org.freedesktop.systemd1.html#)